### PR TITLE
Adds missing information in API document

### LINKED ISSUE #864: Update API documentation
#850 and #839 (same for different branches) makes two RPCAPI methods "disappear" by configuration in the configuration file. This change should be reflected in the API document.

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR documents that affect on the API by the new keys for batch methods added by #850 and #839.

* Adds missing entries in table of contents (`docs/Configuration.md`)
* Several updates to `docs/API.md`:
  * Updates link text for IPv6 address format.
  * Updates specification "add_api_user" method and "add_batch_job" method, respectively.
  * Adds reference to "enable_add_api_user" and "enable_add_batch_job" configuration keys.
  * Adds examples of error messages for both methods.
  * Adds new links in link section.
  * Sorts links in link section.

## Context

This PR resolves #864.

## Changes

Two documents are updated:
* docs/Configuration.md
* docs/API.md

## How to test this PR

Documentation only. Verify the claims in the documents.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -1238,11 +1292,36 @@ A *batch id*.
 
 #### `"error"`
```
Comment (blacksponge) [inline docs/API.md:1293]: Maybe you can add
````markdown
Trying to add a batch when a batch is still running for the current user
```json
{
  "jsonrpc": "2.0",
  "error": {
    "data": {
      "creation_time": "2021-09-27 07:33:40",
      "batch_id": 1
    },
    "code": -32603,
    "message": "Batch job still running"
  },
  "id": 1
}

```
````

Comment (matsduf) [reply docs/API.md:1293]: Good suggestion. Done.

File: docs/API.md [hunk 2]
```diff
@@ -1100,13 +1100,13 @@ An object with the following properties:
...
 ## API method: `add_api_user`
 
-In order to use advanced api features such as the *batch test*, it's necessaire to previously create an api key.
-This key can be obtained with the creation of a user in the system.
-This function allow the creation of a new user and so, the creation of a new api key.
+In order to use the [`add_batch_job`](#API-method-add_batch_job) method an API
+user and key must be created by this method.
 
-Add a new *user*
+This method is not avaialable if [`RPCAPI.enable_add_api_user`] is disabled
```
Comment (mattias-p) [inline docs/API.md:1106]: `avaialable` -> `available`

File: docs/API.md [hunk 3]
```diff
@@ -1145,36 +1145,90 @@ An integer. The value is equal to 1 if the registration is a success, or 0 if it
...
+```
 
 ## API method: `add_batch_job`
 
-Add a run a new *batch test* composed by a set of *domain name* and a *params* object.
+Add a new *batch test* composed by a set of *domain name* and a *params* object.
 All the domains will be tested using identical parameters.
 
-An *api user* can only have one un-finished *batch* at a time.
+This method is not avaialable if [`RPCAPI.enable_add_batch_job`] is disabled
```
Comment (mattias-p) [inline docs/API.md:1227]: `avaialable` -> `available`

File: docs/API.md [hunk 4]
```diff
@@ -1145,36 +1145,90 @@ An integer. The value is equal to 1 if the registration is a success, or 0 if it
...
+(enabled by default).
 
-If an identical *test* for a domain was already enqueued and hasn't been started or was enqueued less than 10 minutes earlier,
-no new *test* is enqueued for this domain.
+API user can be create with the [`add_api_user`](#API-method-add_api_user)
```
Comment (mattias-p) [inline docs/API.md:1230]: `API user` -> `An API user`

`create` -> `created`

File: docs/API.md (hunk 2 above)
Comment (matsduf) [reply docs/API.md]: Fixed.

File: docs/API.md (hunk 4 above)
Comment (matsduf) [reply docs/API.md]: Fixed, but see separate comment on "API user".

**Review by matsduf — COMMENTED** [review]
The method is "add_api_user" but the data type is "username" in the document. I have consistently replace "api user" with "username" and also consistently refer to the key as "api key". I would have been easier if "username" was "api_user" in the method...

**Comment by matsduf** [comment]
@blacksponge, please re-review.

**Review by blacksponge — APPROVED** [review]
looks good to me

**Comment by matsduf** [comment]
@mattias-p, can you re-review?

**Review by mattias-p — APPROVED** [review]

**Pull request merged**